### ISSUE TITLE
Configure memory swapiness for a container 

### ISSUE THREAD

**Issue by washraf** [opencontainers/runtime-tools#111]
How to configure the cgroup memory swapiness using the ocitool or even directly in the config.json file.

**Comment by wking** [comment]
On Fri, Jun 17, 2016 at 02:27:08PM -0700, washraf wrote:
> How to configure the cgroup memory swapiness using the ocitool or
> even directly in the config.json file.

There's no ocitools flag yet, but the setting is here [1].

[1]: https://github.com/opencontainers/runtime-spec/blob/v1.0.0-rc1/config-linux.md#memory

**Comment by washraf** [comment]
"linux": {
                "memory": {
                        "swappiness": 10
                },
                "resources": {
                        "devices": [
                                {
                                        "allow": false,
                                        "access": "rwm"
                                }
                        ]
                },
...
I configured it and It made no effect ??
I test from the memory.swappiness file .

**Comment by wking** [comment]
On Fri, Jun 17, 2016 at 03:05:54PM -0700, washraf wrote:
> "linux": {
>                 "memory": {
>                         "swappiness": 10
>                 },
>                 "resources": {
>                         "devices": [
>                                 {
>                                         "allow": false,
>                                         "access": "rwm"
>                                 }
>                         ]
>                 },
> ...
> I configured it and It made no effect ??
> I test from the memory.swappiness file .

That's probably a runC issue then, although they have tests [1], so
you'll probably want to post very detailed reproduction notes when you
tell them.  Stracing your process might help too.

[1]: https://github.com/opencontainers/runc/blob/v1.0.0-rc1/libcontainer/cgroups/fs/memory_test.go#L274-L298

**Comment by wking** [comment]
This can probably be closed in favor of opencontainers/runc#921.

**Comment by washraf** [comment]
this is also mine .. I have just opened it and I think this should be open to support these kind of configuration in the oci tool it self. Don't you agree ??

**Comment by wking** [comment]
On Fri, Jun 17, 2016 at 03:43:32PM -0700, washraf wrote:
> I think this should be open to support these kind of configuration
> in the oci tool it self. Don't you agree ??

Ah, right.  That's fine with me.

**Comment by hmeng-19** [comment]
#175 makes it possible to specify memory swappiness by `ocitools generate`.

**Comment by kunalkushwaha** [comment]
I guess this issue is resolved. 

using ``ocitools generate --linux-mem-swappiness`` value can be set.

and in config.json ``memory`` section need to be under ``resources`` tag.

``` json
.
.
"linux": {
                 "resources": {
                         "devices": [
                                 {
                                         "allow": false,
                                         "access": "rwm"
                                 }
                         ]
                         },
                        "memory": {
                                 "swappiness": 10
                         },
 ...
```

Can this be closed?

**Comment by hqhq** [comment]
Fixed by #175 .
